src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -415,17 +415,20 @@ def unique(self, level: Hashable | None = None) -> Index:
         See Also
         --------
         unique : Numpy array of unique values in that column.
-        Series.unique : Return unique values of Series object.
+        Series.unique : Return unique values of a Series object.
 
         Examples
         --------
         >>> idx = pd.Index([1, 1, 2, 3, 3])
         >>> idx.unique()
         Index([1, 2, 3], dtype='int64')
         """
-        # TODO: SNOW-1458132 implement unique
-        WarningMessage.index_to_pandas_warning("unique")
-        return Index(self.to_pandas().unique(level=level))
+        # TODO: SNOW-1514782 MultiIndex - support level for Index.unique.
+        return Index(
```

Comment:
let's do this so we don't have to change the existing query compiler interface, and we're not using `reset_index()`:

```suggestion
        return Index(
            data=self._query_compiler.groupby_agg(
                by=self._query_compiler.get_index_names(axis=0), agg_func={}, axis=0, groupby_kwargs={"sort": False, "as_index": True, "dropna": False}, agg_args=[], agg_kwargs={}))
```